gbpservice/neutron/services/grouppolicy/drivers/oc_resource_mapping.py

```
@@ -0,0 +1,271 @@
+from gbpservice.neutron.extensions import group_policy as gp_ext
+
+from gbpservice.neutron.services.grouppolicy.drivers import resource_mapping
+
+LOG = logging.getLogger(__name__)
+
+
+class OCResourceMappingDriver(resource_mapping.ResourceMappingDriver):
+    """Resource Mapping driver for Group Policy plugin.
+
+    This driver implements group policy semantics by mapping group
+    policy resources to various other neutron resources.
+    """
+
+    @log.log
+    def create_policy_target_group_postcommit(self, context):
+        subnets = context.current['subnets']
+        if subnets:
+            l2p_id = context.current['l2_policy_id']
+            l2p = context._plugin.get_l2_policy(context._plugin_context,
+                                                l2p_id)
+            l3p_id = l2p['l3_policy_id']
+            l3p = context._plugin.get_l3_policy(context._plugin_context,
+                                                l3p_id)
+            router_id = l3p['routers'][0] if l3p['routers'] else None
+            for subnet_id in subnets:
+                self._use_explicit_subnet(context._plugin_context, subnet_id,
+                                          router_id)
+        else:
+            self._use_implicit_subnet(context)
+        self._handle_network_service_policy(context)
+        self._handle_policy_rule_sets(context)
+        self._update_default_security_group(context._plugin_context,
+                                            context.current['id'],
+                                            context.current['tenant_id'],
+                                            context.current['subnets'])
+        if (self._is_firewall_in_sc_spec(context, ptg_id=context.current['id'])
+            and self._is_ptg_provider(context, context.current['id'])):
+            self._create_service_vm_sg(context,
+                                       context.current['tenant_id'],
+                                       context.current['id'])
```

What happens if there is no firewall? How do wehandle that case?  Should we raise an exception?